.github/workflows/gradle-check.yml

```diff
@@ -9,12 +9,37 @@ on:
     types: [opened, synchronize, reopened]
 
 jobs:
+  spotless:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Spotless Check
+        run: ./gradlew spotlessCheck
+  precommit:
+    needs: spotless
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: adopt
```

Comment:
`distribution: temurin` please

Reply:
I copy/pasted precommit from [main](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/precommit.yml#L16) which has `adopt` distribution. Updated it.